src/Encamina.Enmarcha.SemanticKernel.Abstractions/ILengthFunctions.cs

```diff
@@ -5,18 +5,54 @@ namespace Encamina.Enmarcha.SemanticKernel.Abstractions;
 /// <inheritdoc/>
 public interface ILengthFunctions : AI.Abstractions.ILengthFunctions
 {
+    /// <summary>
+    /// Dictionary to cache GptEncoding instances based on encoding names.
+    /// </summary>
+    private static readonly Dictionary<string, GptEncoding> EncodingCache = [];
+
+    /// <summary>
+    /// Gets the default GptEncoding for models like `GPT-3.5-Turbo` and `GPT-4` from OpenAI.
+    /// </summary>
+    private static GptEncoding defaultGptEncoding;
+
     /// <summary>
     /// Gets the number of tokens using encodings for models like `GPT-3.5-Turbo` and `GPT-4` from OpenAI on the specified text.
     /// If the text is <see langword="null"/> or empty (i.e., <see cref="string.Empty"/>), returns zero (<c>0</c>).
     /// </summary>
     /// <seealso href="https://platform.openai.com/tokenizer"/>
     /// <seealso href="https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb"/>
-    public static Func<string, int> LengthByTokenCount => (text) => string.IsNullOrEmpty(text) ? 0 : GptEncoding.GetEncoding("cl100k_base").Encode(text).Count;
+    public static Func<string, int> LengthByTokenCount => (text) => string.IsNullOrEmpty(text) ? 0 : GetDefaultGptEncoding().Encode(text).Count;
 
     /// <summary>
     /// Gets the number of tokens using a given encoding on the specified text.
     /// If the text is <see langword="null"/> or empty (i.e., <see cref="string.Empty"/>), returns zero (<c>0</c>).
     /// </summary>
     /// <seealso href="https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb"/>
-    public static Func<string, string, int> LengthByTokenCountUsingEncoding => (encoding, text) => string.IsNullOrEmpty(text) ? 0 : GptEncoding.GetEncoding(encoding).Encode(text).Count;
+    public static Func<string, string, int> LengthByTokenCountUsingEncoding => (encoding, text) => string.IsNullOrEmpty(text) ? 0 : GetCachedEncoding(encoding).Encode(text).Count;
+
+    /// <summary>
+    /// Gets the default GptEncoding instance, creating it if not already initialized.
+    /// </summary>
+    /// <returns>The default GptEncoding instance.</returns>
+    private static GptEncoding GetDefaultGptEncoding()
```

Comment:
As the comment above, I would suggest to remove this method and convert the `defaultGptEncoding` variable in a public read only static member initialize as `GptEncoding.GetEncoding("cl100k_base");`.